### PR TITLE
buildah-fedora: new system container

### PR DESCRIPTION
**Pull request by giuseppe**
Blocked on: https://github.com/projectatomic/atomic/pull/1186

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by giuseppe** [comment]
I experimented with it as part of a comment here: https://github.com/projectatomic/buildah/issues/158

The nice part is that it works as an user container as well, bwrap-oci can run it.

Opening a PR so it doesn't get lost

File: buildah-fedora/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM registry.fedoraproject.org/fedora:27
+
+ENV VERSION=0 RELEASE=1 ARCH=x86_64
+LABEL com.redhat.component="buildah" \
+      atomic.run="once" \
+      atomic.type="system"
+
+RUN mkdir /exports /host && dnf install --setopt=tsflags=nodocs -y buildah && dnf clean all
```
Comment (ashcrow) [inline buildah-fedora/Dockerfile]: nit: It's always nice to have commands on different lines.

**Comment by rhatdan** [comment]
@alexlarsson PTAL

File: buildah-fedora/Dockerfile (hunk 1 above)
Comment (giuseppe) [reply buildah-fedora/Dockerfile]: thanks, I've pushed a new version to split this on multiple lines

**Comment by giuseppe** [comment]
no, we are not able to copy files to the host with `run.once` but if we decide to go this way, we can take advantage of `noContainerService`.

With `noContainerService` we can have a container installed that doesn't require any running systemd service and we can use it to exec into it from the host.

**Comment by ashcrow** [comment]
@giuseppe I'd love that! ❤️

**Comment by alexlarsson** [comment]
Having this rinnande without root is awesome, but Using the vfs backend is kinda lame. Did anyone look at making a backend using the ostree rofiles-fuse helper? Especially with the new copy-up feature. This would be like an non-root overlayfs.

**Comment by rhatdan** [comment]
Not yet we would have to get that added as a storage driver for containers/storage it it would work @giuseppe WDYT?

**Comment by giuseppe** [comment]
@rhatdan I think that is a great idea, I have a PR for adding ostree as option for overlay, but that is completely different than this use case.

We can followup on containers/storage for the new storage driver

**Comment by giuseppe** [comment]
projectatomic/atomic#1186 is approved.  Fine to :ship: this?

**Comment by ashcrow** [comment]
@rh-atomic-bot r+ bbcc66b

**Pull request closed without merging**